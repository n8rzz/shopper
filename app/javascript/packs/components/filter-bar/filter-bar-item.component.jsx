import React from 'react';
import PropTypes from 'prop-types';

export function FilterBarItem(props) {
    return (
        <li>
            <a
                className="filterBar-link"
                href={props.url}
                onClick={props.onChangeFilterHandler}
            >
                {props.text}
            </a>
        </li>
    );
}

FilterBarItem.propTypes = {
    text: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    onChangeFilterHandler: PropTypes.func,
};
