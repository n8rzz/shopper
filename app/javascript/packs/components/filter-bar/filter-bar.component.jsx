import React from 'react';
import PropTypes from 'prop-types';
import { FilterBarItem } from './filter-bar-item.component';

export function FilterBar(props) {
    return (
        <ul className={'filterBar'}>
            {props.items.map((filterBarItem, index) => (
                <FilterBarItem
                    key={`filterBar-item-${index}`}
                    text={filterBarItem.text}
                    url={filterBarItem.url}
                    onChangeFilterHandler={props.onChangeFilterHandler}
                />
            ))}
        </ul>
    );
}

FilterBar.propTypes = {
    items: PropTypes.array.isRequired,
    onChangeFilterHandler: PropTypes.func,
};

FilterBar.defaultProps = {
    onChangeFilterHandler: () => {},
};
