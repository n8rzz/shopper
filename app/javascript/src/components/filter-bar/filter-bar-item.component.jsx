import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import _noop from 'lodash/noop';

export function FilterBarItem(props) {
    const filterBarClassnames = classnames({
        'filterBar-link': true,
        'mix-filterBar-link_isActive': props.isActive,
    });

    return (
        <li>
            <a
                className={filterBarClassnames}
                href={props.url}
                onClick={props.onChangeFilterHandler}
            >
                {props.text}
            </a>
        </li>
    );
}

FilterBarItem.propTypes = {
    isActive: PropTypes.bool.isRequired,
    text: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    onChangeFilterHandler: PropTypes.func,
};

FilterBarItem.defaultProps = {
    onChangeFilterHandler: _noop,
};
