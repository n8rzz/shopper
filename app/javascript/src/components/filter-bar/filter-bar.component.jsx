import React from 'react';
import PropTypes from 'prop-types';
import _noop from 'lodash/noop';
import { FilterBarItem } from './filter-bar-item.component';

export class FilterBar extends React.PureComponent {
    constructor(props) {
        super(props);

        this._onChangeFilterHandler = this._onChangeFilter.bind(this);
        this.state = {
            activeFilter: props.initialSelection,
        };
    }

    componentWillUnmount() {
        this._onChangeFilterHandler = null;
    }

    _buildFilterBarListItemsJsx() {
        return this.props.items.map((filterBarItem, index) => {
            const isActive = this.state.activeFilter === filterBarItem.text;

            return (
                <FilterBarItem
                    key={`filterBar-item-${index}`}
                    isActive={isActive}
                    text={filterBarItem.text}
                    url={filterBarItem.url}
                    onChangeFilterHandler={this._onChangeFilterHandler}
                />
            );
        });
    }

    _onChangeFilter(event) {
        this.setState({ activeFilter: event.target.text }, this.props.onChangeFilterHandler(event));
    }

    render() {
        const filterbarListItemsJsx = this._buildFilterBarListItemsJsx();

        return (
            <ul className={'filterBar'}>
                {filterbarListItemsJsx}
            </ul>
        );
    }
}

FilterBar.propTypes = {
    items: PropTypes.array.isRequired,
    initialSelection: PropTypes.string.isRequired,
    onChangeFilterHandler: PropTypes.func,
};

FilterBar.defaultProps = {
    onChangeFilterHandler: _noop,
};
