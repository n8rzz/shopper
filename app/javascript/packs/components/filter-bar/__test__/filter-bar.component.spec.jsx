import React from 'react';
import { shallow } from 'enzyme';
import { FilterBar } from '../filter-bar.component';

const filterBarItemListMock = [
    {
        text: 'Alpha',
        url: '/items',
    },
    {
        text: 'Department',
        url: '/items/?order=department',
    },
];

describe('FilterBar', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<FilterBar
            initialSelection={filterBarItemListMock[0].text}
            items={filterBarItemListMock}
        />);

        expect(component).toMatchSnapshot();
    });
});
