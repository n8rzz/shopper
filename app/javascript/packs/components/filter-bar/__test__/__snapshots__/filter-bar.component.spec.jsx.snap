// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar renders correctly with valid props 1`] = `
<ul
  className="filterBar"
>
  <FilterBarItem
    isActive={true}
    key="filterBar-item-0"
    onChangeFilterHandler={[Function]}
    text="Alpha"
    url="/items"
  />
  <FilterBarItem
    isActive={false}
    key="filterBar-item-1"
    onChangeFilterHandler={[Function]}
    text="Department"
    url="/items/?order=department"
  />
</ul>
`;
